### PR TITLE
tyr: tyr task to import bounding shape

### PR DESCRIPTION
**Pull request by TeXitoi**

### CODE REVIEW

File: source/tyr/tyr/tasks.py [hunk 1]
```diff
@@ -292,6 +295,83 @@ def cities(osm_path):
...
+    """ Set the bounding shape to a custom value """
+    shape = None
+    if shape_path.endswith(".poly"):
+        with open (shape_path, "r") as myfile:
+            shape = parse_poly(myfile.readlines())
+    elif shape_path.endswith(".wkt"):
+        with open (shape_path, "r") as myfile:
+            shape = wkt.loads(myfile.read())
+    else:
+        logging.error("bounding_shape: {} has an unknown extension.".format(shape_path))
```
Comment (antoine-de) [inline source/tyr/tyr/tasks.py:358]: you need to exit there non ?

File: source/tyr/tyr/tasks.py [hunk 2]
```diff
@@ -292,6 +295,83 @@ def cities(osm_path):
...
+
+        http://wiki.openstreetmap.org/wiki/Osmosis/Polygon_Filter_File_Format
+    """
+    in_ring = False
+    coords = []
+
+    for (index, line) in enumerate(lines):
+        if index == 0:
+            # first line is junk.
+            continue
```
Comment (antoine-de) [inline source/tyr/tyr/tasks.py:312]: useless continue

File: source/tyr/tyr/tasks.py [hunk 3]
```diff
@@ -292,6 +295,83 @@ def cities(osm_path):
...
+    elif shape_path.endswith(".wkt"):
+        with open (shape_path, "r") as myfile:
+            shape = wkt.loads(myfile.read())
+    else:
+        logging.error("bounding_shape: {} has an unknown extension.".format(shape_path))
+
+    conf = load_instance_config(instance_name)
+    connection_string = "postgres://{u}:{pw}@{h}/{db}"\
+        .format(u=conf.pg_username, pw=conf.pg_password, h=conf.pg_host, db=conf.pg_dbname)
+    print(connection_string)
```
Comment (antoine-de) [inline source/tyr/tyr/tasks.py]: old log

File: source/tyr/tyr/tasks.py (hunk 2 above)
Comment (TeXitoi) [inline source/tyr/tyr/tasks.py:312]: That's a pure copy and paste from the given address, I prefere not change it if not necessary for future evolution.

File: source/tyr/tyr/tasks.py [hunk 4]
```diff
@@ -292,6 +295,83 @@ def cities(osm_path):
...
+    engine = sqlalchemy.create_engine(connection_string)
+    # create the line if it does not exists
+    engine.execute("""
+    INSERT INTO navitia.parameters (shape)
+    SELECT NULL WHERE NOT EXISTS (SELECT * FROM navitia.parameters)
+    """).close()
+    # update the line
+    engine.execute("UPDATE navitia.parameters SET shape_computed = FALSE, shape = '{shape}'"\
```
Comment (antoine-de) [inline source/tyr/tyr/tasks.py]: don't we need then a simplify ? (maybe a param to the tyr command ?)

**Pull request merged**